lib/CollectionManager.js

```diff
@@ -25,11 +27,28 @@ class CollectionManager {
   }
 
   async dumpCollection() {
-    const query = this.dumpProgress.count
+    const todoQuery = this.dumpProgress.count
       ? { _id: { $gt: this.dumpProgress.token } }
       : {};
-    let cursor = this.collection.find(query);
-    const count = (await cursor.count()) + this.dumpProgress.count;
+    let cursor = this.collection.find(todoQuery).batchSize(BATCH_SIZE);
+    const totalCount = (await cursor.count()) + this.dumpProgress.count;
+
+
+    if (!await cursor.hasNext() && this.dumpProgress.completeDate) {
+      logger.info(`${this.collectionName} already finished on ${this.dumpProgress.completeDate}`);
+      return;
+    }
+
+    if (!await cursor.hasNext() && !this.dumpProgress.completeDate) {
+      await this.dumpProgress.complete();
+      logger.info(`${this.collectionName} has no more to do, marking as finished on ${this.dumpProgress.completeDate}`);
```

Comment:
How about just `logger.info(\`${this.collectionName} dump completed at ${this.dumpProgress.completeDate}\`);`

Reply:
I want to distinguish it from the case when a dump was actually processed and then 'completed' vs just started and then discovered that there was no work to do.  I'm open to other verbiage suggestions though.